### PR TITLE
Siamese Network Classifier for ISIC 2020 Dataset (Nicholas Lany)

### PR DESCRIPTION
**Pull request by nicklany01**
Siamese network classifier used to classify images from the ISIC 2020 Dataset. The aim is to train a siamese network which can distinguish similar and dissimilar images, and using the model that was used to distinguish features to classify images into benign or malignant melanoma classifications. The goal of this model is to achieve an accuracy of around 80% on the test set, and it achieved an accuracy of 99.1% after 25 epochs training on 50,000 pairs of images for each epoch.

### CODE REVIEW

**Comment by hanemma7moud** [comment]
This is an initial inspection, no action is required at this point
Recognition Problem : total : 17

Solves problem: The solution is appropriate for the problem but no performance to evaluate unless the reported training (99.1%) and testing (80%) accuracy with overfitting. (**3**)


Implementation functions as intended (2)

Good design: Well-designed (1)

Commenting: Clear and sufficient comments throughout the code. (1)

Difficulty: Hard (10)

Note:
The resulting manifold is not generated and confusion matrix.
The targeted accuracy is acheived but I can not evaluate the model without assessing other performance metrics especially with an imbalance datatset.
Lack of Data Diversity: Resampling merely duplicates existing samples without adding new information, increasing the risk of overfitting, especially with small, diverse minority classes. In medical imaging, data diversity is essential for the model to learn robust, generalizable features.
Merge

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. </td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Doc strings smissing.</td><td>-1</td></tr>
	<tr><td></td><td>Data leakage found. Has data splitting. Lacks data diversity.</td><td>-2</td></tr>
	<tr><td></td><td>Difficulty : Hard.  Siamese network (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-10</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Missing manifold plot etc. -2

**Comment by shakes76** [comment]
Cant merge because of conflicting changes to main repo files (PDF file). Please update for merge, doesn't affect grade.
